src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferDictionaryTryAddValueOverGuardedAddTests.cs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Globalization;
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferDictionaryTryAddValueOverGuardedAddAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Performance.CSharpPreferDictionaryTryAddValueOverGuardedAddFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.PreferDictionaryTryAddValueOverGuardedAddAnalyzer,
+    Microsoft.NetCore.VisualBasic.Analyzers.Performance.BasicPreferDictionaryTryAddValueOverGuardedAddFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class PreferDictionaryTryAddValueOverGuardedAddTests
+    {
+
+        #region C# Tests
+
+        private const string CSharpTemplate = @"
+using System;
+using System.Collections.Generic;
+
+namespace UnitTests {{
+    public class Test {{
+        public void M() {{
+            var d = new Dictionary<int, int>();
+            {0}
+        }}
+    }}
+
+    public class FakeDictionary<TKey, TValue> {{
+        public bool ContainsKey(TKey key) => throw null;
+        public void Add(TKey key, TValue value) => throw null;
+    }}
+}}";
+
+        private const string CheckNegativeReturnedValue = @"
+if (!{|#0:d.ContainsKey(5)|})
+{
+	{|#1:d.Add(5, 6)|};
+}";
```

Comment:
Is there a test where `ContainsKey` and `Add` are invoked on two different dictionaries?

Reply:
There is now :)